lib/AssemblyUtil/FastaToAssembly.py

```diff
@@ -473,6 +482,12 @@ def _validate_mass_params(self, params):
                 raise ValueError(f"Missing {_ASSEMBLY_NAME} field in {_INPUTS} entry #{i}")
         self._get_int(params.get(_MCL), f"If provided, {_MCL}", minimum=2)
 
+        # validate object_metadata is a mapping if provided
+        obj_metas = [i.get(_OBJ_META) for i in inputs]
+        for obj_meta in obj_metas:
+            if obj_meta and type(obj_meta) != dict:
```

Comment:
```suggestion
            if obj_meta is not None and type(obj_meta) != dict:
```
Otherwise lists will pass this check.
Do you want to check that the keys & values are str/str as well? The workspace will catch it if not

Reply:
👍 added str type check.

Reply:
Actually now I think about it nulls might be allowed for values as well. Can you double check?

Reply:
According to WS spec, it needs to be str/str pair. I am not sure if the WS code base removes null values. 
https://github.com/kbase/workspace_deluxe/blob/develop/workspace.spec#L85-L88

Reply:
oh by the way, why lists will pass the check I do `if obj_meta and type(obj_meta) != dict:`?

Reply:
> According to WS spec, it needs to be str/str pair. I am not sure if the WS code base removes null values.

It could, and I'm pretty sure that in some cases it does allow nulls where a string is specified, because there's no way to type `string | null` in the KIDL. In this case though, nulls are not allowed:
https://github.com/kbase/workspace_deluxe/blob/develop/src/us/kbase/workspace/database/WorkspaceUserMetadata.java#L57-L59

> oh by the way, why lists will pass the check I do if obj_meta and type(obj_meta) != dict:?

Because if `obj_meta == []` then `if obj_meta` is false, so the rest of the conditional will be short circuited since it's an `and` and the exception won't be thrown.

Reply:
why need to explicitly check `obj_meta is not None`? @MrCreosote

Reply:
Anyway, nothing more to do here so :+1: